playground/examples/svelte-change-tracking/editor.svelte

```diff
@@ -0,0 +1,64 @@
+<script lang="ts">
+import 'prosekit/basic/style.css'
+
+import { Themes } from '@prosekit/themes'
+import type { NodeJSON } from 'prosekit/core'
+import { CommitRecorder, type Commit } from 'prosekit/extensions/commit'
+
+import EditorDiff from './editor-diff.svelte'
+import EditorMain from './editor-main.svelte'
+
+let commits: { id: string; date: Date; commit: Commit }[] = []
+let key = 0
+let defaultDoc: NodeJSON
+
+const commitRecorder = new CommitRecorder()
+
+function handleCommit() {
+  const commit = commitRecorder.commit()
+  if (!commit) return
+  const id = Math.random().toString(36).slice(2, 9)
+  commits = [{ id, date: new Date(), commit }, ...commits]
+}
+
+function handleRestore(id: string) {
+  const index = commits.findIndex((commit) => commit.id === id)
+  if (index === -1) return
+  defaultDoc = commits[index].commit.doc
+  commits = commits.slice(index)
+  key += 1
+}
+</script>
+
+<div class="grid grid-cols-2 gap-2">
+  <div class="flex flex-col gap-4">
+    <div class="max-h-md">
+      {#key defaultDoc}
```

Comment:
I don't understand Svelte enough. Between `{#key defaultDoc}` (an object as the key) and `{#key key}` (a number), which one is better? We can delete the `key` variable (in line 12) if we choose `{#key defaultDoc}`.

Reply:
So, from the docs:

> Key blocks destroy and recreate their contents when the value of an expression changes.

Ergo, my thinking was, "what changes, and as a result, should make us re-render the editor with the new content"? From a simple point of view, it's the `defaultDoc` (content of the editor) that changes, and should cause a re-render.

But yes, I also believe the `key` variable can be a tool to use when re-rendering the editor. Matter of fact, if the `key` variable is already being used elsewhere, we can use it instead of `defaultDoc`.

Wdyt?